GetStream/stream-chat-android
# Removed Icons dependency, added RTL support to some icons and text input

### 🎯 Goal

We needed to review our components and make sure we support RTL for our users, similar to how XML does.

### 🛠 Implementation details

Mostly updating icons and drawables, pulling out the Icons dependency from our SDK. Added Content based layout direciton to input field.

### 🎨 UI Changes

| Before | After |
| --- | --- |
| ![Screenshot_1643213098](https://user-images.githubusercontent.com/17215808/151200090-5b0856e8-4579-4eeb-946d-2f593031602f.png) | ![Screenshot_1643213020](https://user-images.githubusercontent.com/17215808/151200158-a957adbd-f038-4afd-aeec-f980e81e3c9f.png) |

Channels
| XML | Compose |
| --- | --- |
| ![Screenshot_1643213009](https://user-images.githubusercontent.com/17215808/151200297-edf43919-4116-415f-bf4e-d2c015d85ac5.png) | ![Screenshot_1643212997](https://user-images.githubusercontent.com/17215808/151200332-fdc47e49-9b03-4edf-9555-22cadda70543.png) |

Messages
| XML | Compose |
| --- | --- |
| ![Screenshot_1643213025](https://user-images.githubusercontent.com/17215808/151200474-09fdf90c-c7dc-4950-8260-a998e8588a4b.png) | ![Screenshot_1643213020](https://user-images.githubusercontent.com/17215808/151200499-2639ec07-819e-4d83-b9a6-892ede36e3ff.png) |

### 🧪 Testing

1. Run both XML and Compose apps side-by side.
2. Turn on an RTL based language (Arabic works).
3. Make sure to check that both of the apps look more less the same (structurally). There are some UI differences obviously but the core part of the structure, direction, placement etc should be the same.

### ☑️Contributor Checklist

#### General

#### Code & documentation

### ☑️Reviewer Checklist
